pkg/controller/dns/external_dns.go

```diff
@@ -91,24 +91,37 @@ func instances(conf *config.Config) []instance {
 	}
 }
 
-func publicConfig(conf *config.Config) *manifests.ExternalDnsConfig {
-	return &manifests.ExternalDnsConfig{
-		TenantId:           conf.TenantID,
-		Subscription:       conf.PublicZoneConfig.Subscription,
-		ResourceGroup:      conf.PublicZoneConfig.ResourceGroup,
-		Provider:           manifests.PublicProvider,
-		DnsZoneResourceIDs: util.Keys(conf.PublicZoneConfig.ZoneIds),
-	}
+func publicConfigForIngress(conf *config.Config) *manifests.ExternalDnsConfig {
+	return manifests.NewExternalDNSConfig(
+		conf,
```

Comment:
we are passing in so many arguments here. should use a struct as input instead, it's too long.

Reply:
resolving per offline discussion re: correct usage of params

Reply:
so we'd have one externaldnsconfig that's visible to the controllers and one internally? that just seems like a confusing pattern.

Either we declare the type in the controller, which would create a cyclical dependency, or we expose the fields in the public type - but that violates the initial premise of forming the constructor (ie to prevent incorrect usage/assignment of the fields).

We could also create a third type within manifests but that seems excessive